data-prepper-core/src/test/java/com/amazon/dataprepper/server/DataPrepperServerTest.java

```diff
@@ -50,6 +58,76 @@ public void testScrapeFailure() throws IOException, InterruptedException, URISyn
         dataPrepperServer.stop();
     }
 
+    @Test
+    public void testListPipelines() throws URISyntaxException, IOException, InterruptedException {
+        setRegistry(new PrometheusRegistryMockScrape(PrometheusConfig.DEFAULT, ""));
+        final DataPrepper mockDataPrepper = Mockito.mock(DataPrepper.class);
+        final String pipelineName = "testPipeline";
+        Mockito.when(mockDataPrepper.getTransformationPipelines()).thenReturn(
+                Collections.singletonMap(pipelineName, Mockito.mock(Pipeline.class))
+        );
+        final DataPrepperServer dataPrepperServer = new DataPrepperServer(8080, mockDataPrepper);
+        dataPrepperServer.start();
+
+        HttpRequest request = HttpRequest.newBuilder(new URI("http://127.0.0.1:8080/list"))
+                .GET().build();
+        HttpResponse response = HttpClient.newHttpClient().send(request, HttpResponse.BodyHandlers.ofString());
+        final String expectedResponse = OBJECT_MAPPER.writeValueAsString(
+                Collections.singletonMap("pipelines", Arrays.asList(
+                        Collections.singletonMap("name", pipelineName)
+                ))
+        );
+        Assert.assertEquals(200, response.statusCode());
+        Assert.assertEquals(expectedResponse, response.body());
+        dataPrepperServer.stop();
```

Comment:
nit: Similar comment about common code - could this be brought up into an `@After` block to ensure shutdown? Or at the very least put in a finally block?

Reply:
Hmm yeah, so the stop() function can be moved into a finally.